### PR TITLE
Feature add rtttl frame

### PR DESCRIPTION
**Pull request by PatrickDekker98**
an rtttl frame that sends an rtttl string was added :/

### CODE REVIEW

**Review by itzandroidtab — CHANGES_REQUESTED** [review]
As this is a string this can use the string optimatisation

File: code/headers/frame_types.hpp [hunk 1]
```diff
@@ -550,6 +551,16 @@ namespace r2d2 {
     };
 
     /*
+    * This is a frame that will be send to the sound module. 
+    * It contains a simple rtttl string 
+    * this was not my idea either 
+    */
+    R2D2_PACK_STRUCT
+    struct frame_rtttl_string_s {
+        //the rtttl string to be send
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: This comment needs a space between // and the

File: code/headers/frame_types.hpp [hunk 2]
```diff
@@ -550,6 +551,16 @@ namespace r2d2 {
     };
 
     /*
+    * This is a frame that will be send to the sound module. 
+    * It contains a simple rtttl string 
+    * this was not my idea either
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: This part doesnt add anything to the docs

**Comment by LRstudentHU** [comment]
> 
> 
> As this is a string this can use the string optimatisation

This is actually required before merging. Please look at the characters frame from display for an example.

File: code/headers/frame_types.hpp (hunk 2 above)
Comment (LRstudentHU) [reply code/headers/frame_types.hpp]: Yup, please remove it before merging

**Review by LRstudentHU — CHANGES_REQUESTED** [review]

File: code/headers/frame_types.hpp [hunk 3]
```diff
@@ -628,6 +629,16 @@ namespace r2d2 {
...
     /*
+    * @brief This is a frame that will be send to the sound module. 
+    * @details It contains a simple rtttl string 
+    * this was not my idea either 
+    */
+    R2D2_PACK_STRUCT
+    struct frame_rtttl_string_s {
+        // the rtttl string to be send 
+        char rtttl_string[248];
+    };
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp:632]: Can you add a newline after your struct?

File: code/headers/frame_types.hpp (hunk 2 above)
Comment (itzandroidtab) [reply code/headers/frame_types.hpp]: You still didnt remove the "this was not my idea either". And you can remove the `@detail` and ` @brief` as the rest of the file doesnt use them ether.

File: code/headers/frame_types.hpp [hunk 4]
```diff
@@ -764,6 +775,12 @@ namespace r2d2 {
     )
 
     R2D2_INTERNAL_FRAME_HELPER(frame_temperature_s, TEMPERATURE)
+    
+    R2D2_INTERNAL_FRAME_HELPER(
+            frame_rtttl_string_s, 
+            RTTTL_STRING,
+            R2D2_OPTIMISE_STRING(frame_rtttl_string_s, rtttl_string)
+        )
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: This has 8 spaces. Could you change it to 4?

Comment (PatrickDekker98) [reply code/headers/frame_types.hpp]: should be resolved

File: code/headers/frame_types.hpp (hunk 3 above)
Comment (PatrickDekker98) [reply code/headers/frame_types.hpp:632]: yes yes should be resolved

File: code/headers/frame_types.hpp [hunk 5]
```diff
@@ -628,6 +629,16 @@ namespace r2d2 {
     };
 
     /*
+    * @brief This is a frame that will be send to the sound module.
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: The `@brief` and `@details` should be removed to conform the rest of the file

File: code/headers/frame_types.hpp [hunk 6]
```diff
@@ -628,6 +629,16 @@ namespace r2d2 {
     };
 
     /*
+    * @brief This is a frame that will be send to the sound module. 
+    * @details It contains a simple rtttl string 
+    */
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp:627]: This struct is missing a wiki link

Comment (PatrickDekker98) [reply code/headers/frame_types.hpp:627]: should be resolved

File: code/headers/frame_types.hpp (hunk 5 above)
Comment (PatrickDekker98) [reply code/headers/frame_types.hpp]: should be resolved

**Review by itzandroidtab — APPROVED** [review]

**Review by LRstudentHU — APPROVED** [review]

**Pull request merged**